mtags/src/main/scala-2/scala/meta/internal/pc/AutoImports.scala

```diff
@@ -47,7 +47,7 @@ trait AutoImports { this: MetalsGlobal =>
     findLastVisitedParentTree(pos) match {
       case Some(_: Import) => None
       case _ =>
-        val enclosingPackage = lastVisitedParentTrees.collectFirst {
+        val enclosingPackage = lastVisitedParentTrees.reverse.collectFirst {
```

Comment:
I wonder if we can make it a bit more efficient than using reverse to avoid creating a new list :thinking: 

Maybe we could create an extension for list in MtagsEnrichments?

Something like:

```scala
implicit class XtensionList[T](list: List[T]) {
  def collectLast[R](fun: PartialFunction[T, R]): Option[R] = {
    def loop(current: List[T]): Option[R] = {
      current match {
        case Nil => None
        case head :: tl =>
          loop(tl).orElse(fun.lift(head))
      }
    }
    loop(list)
  }
}
```

Reply:
yes sure! could be useful and more readable